x/evm/state/log.go

```diff
@@ -1,53 +1,27 @@
 package state
 
 import (
-	"encoding/json"
-
 	"github.com/ethereum/go-ethereum/common"
 	ethtypes "github.com/ethereum/go-ethereum/core/types"
-	"github.com/sei-protocol/sei-chain/x/evm/types"
 )
 
 type Logs struct {
 	Ls []*ethtypes.Log `json:"logs"`
 }
 
 func (s *DBImpl) AddLog(l *ethtypes.Log) {
-	// TODO: potentially decorate log with block/tx metadata
-	store := s.k.PrefixStore(s.ctx, types.TransientModuleStateKey(s.ctx))
-	logs := Logs{Ls: []*ethtypes.Log{}}
-	ls, err := s.GetAllLogs()
-	if err != nil {
-		s.err = err
-		return
-	}
-	logs.Ls = append(ls, l)
-	logsbz, err := json.Marshal(&logs)
-	if err != nil {
-		s.err = err
-		return
-	}
-	store.Set(LogsKey, logsbz)
+	s.logs = append(s.logs, l)
```

Comment:
Do we need any concurrency control on these functions?

Reply:
should be fine since there will be one statedb instance per tx, and there is no parallelization within the tx